src/DataTypes/array.tsx

```diff
@@ -0,0 +1,161 @@
+import type { JSONSchema7 as JSONSchema } from 'json-schema';
+import {
+	FULL_TEXT_SLUG,
+	createModelFilter,
+	type FormData,
+} from '../components/Filters/SchemaSieve';
+import { CreateFilter, getDataTypeSchema } from './utils';
+import { getDataModel } from '.';
+import { isJSONSchema, getRefSchema } from '../AutoUI/models/helpers';
+
+export const operators = () => ({
+	contains: 'contains',
+	not_contains: 'does not contain',
+});
+
+export type OperatorSlug =
+	| keyof ReturnType<typeof operators>
+	| typeof FULL_TEXT_SLUG;
+
+const isSchemaWithPrimitiveItems = (
+	schema: JSONSchema,
+): schema is JSONSchema & { items: JSONSchema } =>
+	!!schema?.items &&
+	typeof schema.items !== 'boolean' &&
+	'type' in schema.items;
+
+const isArrayOfObjectSchema = (schema: JSONSchema): boolean => {
+	return (
+		!!schema?.items &&
+		typeof schema.items !== 'boolean' &&
+		'properties' in schema.items &&
+		!!schema.items.properties
+	);
+};
+
+const buildFilterForPropertySchema = (
+	field: string,
+	operator: OperatorSlug,
+	value: string,
+	schema: JSONSchema,
+): JSONSchema | {} => {
```

Comment:
> {}

I don't think our linter likes this, it should usually be `object` or `Record<string, any>` or something else I don't recall